service_discovery_manager/Project/Android/app/src/main/cpp/server_runner_jni.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2019 Samsung Electronics Co., Ltd
+ *
+ * Licensed under the Flora License, Version 1.1 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://floralicense.org/license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "server_runner_jni.h"
+
+#include <android/log.h>
+#include <jni.h>
+
+#include "server_runner.h"
+
+static JavaVM* g_jvm = nullptr;
+static jobject g_class_loader = nullptr;
+static jmethodID g_find_class_method_id = 0;
+static ServerRunner* g_server_runner = nullptr;
+
+static const char* const kClassName = "app/samsung/org/servicediscovery/SDServerService";
+static const char* const kLogTag = "SERVICE-DISCOVERY";
+
+int Java_startChromeRenderer() {
+  __android_log_print(ANDROID_LOG_DEBUG, kLogTag, "Start Chrome as renderer");
+
+  if (!g_jvm || !g_class_loader || !g_find_class_method_id) {
+    __android_log_print(ANDROID_LOG_ERROR, kLogTag, "Not ready to call Java method");
+    return -1;
+  }
+
+  JNIEnv* env;
+  if (g_jvm->GetEnv(reinterpret_cast<void**>(&env), JNI_VERSION_1_6) != JNI_OK) {
+    // TODO(yh106.jung): DetachCurrentThread should be call at the end of thread.
+    if (g_jvm->AttachCurrentThread(&env, NULL) != JNI_OK) {
+        __android_log_print(ANDROID_LOG_ERROR, kLogTag, "GetEnv failed");
+        return -1;
+    }
+  }
+
+  auto clazz = static_cast<jclass>(env->CallObjectMethod(g_class_loader, g_find_class_method_id, env->NewStringUTF("app/samsung/org/servicediscovery/SDServerService")));
```

Comment:
kClassName instead of "app/samsung/org/servicediscovery/SDServerService"?

Reply:
Done